### PR TITLE
fix: allow innerHTML to replace non-empty node, credit @tokichie

### PR DESCRIPTION
**Pull request by iambumblehead**
Copy-pasted changes from @tokichie https://github.com/snabbdom/snabbdom/pull/347 re-applied to the current master branch

### CODE REVIEW

**Comment by iambumblehead** [comment]
happy new year bump!

File: src/init.ts [hunk 1]
```diff
@@ -138,7 +138,9 @@ export function init(
     return function rmCb() {
       if (--listeners === 0) {
         const parent = api.parentNode(childElm) as Node;
-        api.removeChild(parent, childElm);
+        if (parent) {
```
Comment (paldepind) [inline src/init.ts:141]:
```suggestion
        if (parent !== null) {
```

Just to be more explicit about what is checked here.

**Comment by paldepind** [comment]
LGTM, except for the one nitpick comment :)

File: src/init.ts (hunk 1 above)
Comment (iambumblehead) [reply src/init.ts]: thank you, the suggested change is amended to the commit

**Comment by paldepind** [comment]
Thanks!

**Pull request merged**